[Marcus Lee] iP

marclzh opened the pull request:
# Elp Chatbot
*The chatbot providing all your deadline needs!*

### Elp can help you  
- Track your tasks
- Save your tasks to a convenient text file
- Many more...

### Elp is
- [x] Helpful
- [ ] Unique
- [x] Easy to use!

### All you need to do is 
1. Download from [here](https://github.com/marclzh/ip)
2. Add your tasks
3. ???
4. Profit! 💰 💰 

### Features
- Managing Todos
- Managing Deadlines 
- Managing Events
- More to come!

### Taking a peek at the code `public static void main(String[] args)`.
```java {.line-numbers}
public static void main(String[] args) {
    new Duke().run();
}
```
### Look at all these fantastic reviews!
> Fantastic - Anonymous Reviewer
> Provided all the help I needed! - Enthusiastic Reviewer

Commit:
Add Gradle support

marclzh committed:
- Updated .gitignore to ignore .class files.

marclzh committed:
Finished functionality for Level-1

marclzh committed:
Completed Level-2 Functionality

marclzh committed:
Completed Level-3 Functionality

marclzh committed:
Completed Level-4 functionality

marclzh committed:
Completed A-TextUiTesting

marclzh committed:
Completed Level-5 Functionality

marclzh committed:
Updated Expected Test File

marclzh committed:
Completed Level-6 functionality

marclzh committed:
Updated Imports

marclzh committed:
Added Enums for consistency with commands inputted.

marclzh committed:
Add Enums to different java class

marclzh committed:
Add save to harddrive functionality (Level-7)

marclzh committed:
Add date formatting for tasks (Level-8)

marclzh committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	.gitignore
#	src/main/java/Duke.java

marclzh committed:
Refactor code to suit OOP style better

marclzh committed:
Sort classes into packages

marclzh committed:
Update .bat test to work with packages

marclzh committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

marclzh committed:
Gradle fix

marclzh committed:
Add JUnit Test Cases

marclzh committed:
Create JAR file

marclzh committed:
Fix + add JavaDocs

marclzh committed:
Adjust code to follow coding standard

marclzh committed:
Add find functionality (Level-9)

marclzh committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/exception/DukeException.java
#	src/main/java/duke/task/ToDo.java

marclzh committed:
Merge branch 'branch-Level-9'

marclzh committed:
Bugfix mark harddisk saving

Bubbl3T commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,45 @@
+     *
+     * @param command input command
+     */
+    Command(String command) {
```
You may want to add the `public` access modifier here to be clearer with your code.

Bubbl3T commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,45 @@
+     * @param command input command
+     */
+    Command(String command) {
+        this.inputCommand = command;
```
I think `this.` is unnecessary here, since the variable `inputCommand` does not get overshadowed by the `command` variable.

Bubbl3T commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,87 @@
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
```
You might want to import the `java.` classes before the `duke.` classes so that your code pass the StyleCheck 😉

Bubbl3T commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,149 @@
+     *
+     * @param tasks task list.
+     */
+    public static void Load(List<Task> tasks) {
```
I think using lower case for the first character of your method name will be better.

jjtan444 replied:
I agree with other commenter, method names should be written in camelCase. I noticed the same issue in your `Save` method too.

Bubbl3T commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,149 @@
+     * @param tasks task list.
+     * @param isDeleted if the contents of the file should be deleted and reset.
+     */
+    public static void Save(ArrayList<Task> tasks, boolean isDeleted) {
```
I think using lower case for the first character of your method name will be better.

Bubbl3T commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,149 @@
+    /**
+     * Loads the data from the file found in the saved location into the task list.
+     *
+     * @param tasks task list.
```
You might want to capitalize the parameter description for `tasks`.
i.e. `@param tasks Task list.`

Bubbl3T commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,149 @@
+            LocalDateTime due = LocalDateTime.parse(taskDue, Task.OUTPUT_DATE_FORMAT);
+            task = new Deadline(taskDescription, due);
+            break;
+
```
You might want to add a `default` case to deal with potential errors that can occur.

Bubbl3T commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,128 @@
+        if (!tasks.isEmpty()) {
+            System.out.printf("Here's all I could find with \"%s\"!\n", match);
+            for (int i = 1; i <= tasks.size(); i++) {
+                Task temp = tasks.get(i - 1);
```
It might be better to rename `temp` to `task` to be clearer with what the object is.

Bubbl3T commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,69 @@
+        }
+        String description = temp[0];
+        String dueDate = temp[1];
+        LocalDateTime deadline;
```
You might want to rename `deadline` with `dateTime` instead for better clarity.

Bubbl3T commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,69 @@
+            throw new DukeException("-Deadline- Please follow the format of ~description~ /by dd-MM-yyyy HHmm!\n");
+        }
+        String description = temp[0];
+        String dueDate = temp[1];
```
It might be better to rename `dueDate` to `dateString` just so that readers do not confuse it with the `dueDate` stored in the Deadline object.

Bubbl3T commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,69 @@
+ */
+public class Event extends Task {
+    /* Duration field */
+    private LocalDateTime duration;
```
Perhaps `dateTime` might be better than `duration` as a variable name here? I think `duration` might suggest that there is a start and end time, or maybe the number of minutes of the event instead.

Bubbl3T commented on src/main/java/duke/task/ToDo.java:
```diff
@@ -0,0 +1,37 @@
+package duke.task;
+
```
You might want to remove extra blank likes here to make your code look more clean :)

Bubbl3T commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,69 @@
+package duke.task;
+
+import duke.exception.DukeException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
```
You might want to import the `java.` classes before the `duke.` classes so that your code pass the Style Check later on. 😉

Bubbl3T reviewed (commented):
I like your code, it is very clean and easy to read. 👍🏻

jjtan444 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,128 @@
+            if (markStatus == Command.MARK) {
+                currTask.markDone();
+                Storage.Save(tasks, true);
+                System.out.println("Task successfully marked!");
```
Perhaps you might want to consider using the `UI` class to output messages to user.

jjtan444 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,69 @@
+     * @throws DukeException if no task or incorrect formatting is given.
+     */
+    public static Deadline createDeadline(String in) throws DukeException {
+        String[] temp = in.split(" */by* ");
```
You might want to rename `temp` to provide more information about the content of the variable. Do consider fixing similar issues in other classes as well.

jjtan444 reviewed (commented):
Overall looks well done with only very minor issues. There are some design implementations which I can learn from as well. 👍

marclzh committed:
Add checkstyle and fix style changes

marclzh committed:
Add basic gui capabilities

marclzh committed:
Completed GUI functionality and resolved review issues

marclzh committed:
Add GUI changes

marclzh committed:
Add assertions to Duke

marclzh committed:
Adjust assertions

marclzh committed:
Refactor and cleanup code

marclzh committed:
Merge branch 'A-Assertions'

marclzh committed:
Set up CI

marclzh committed:
Debug CI

marclzh committed:
Add assertion for pull request

marclzh committed:
Refactor parsing of date

marclzh committed:
Merge pull request #1 from marclzh/A-CodeQuality

Refactor parsing of date

marclzh committed:
Merge pull request #2 from marclzh/A-Assertions

Add assertions

marclzh committed:
Fix forward reference in Storage class

marclzh committed:
Remove text-ui tests from CI

marclzh committed:
Add Priority Feature

Priority tracking is not currently implemented in Elp.

A priority tracking feature can help users better prioritise and
keep track of tasks.

Adding a priority field to the tasks ToDo, Event and Deadline
allows for the priority to be displayed as well as the other
relevant fields to be displayed.

Let's add the possible priorities to a Priority Enum class and
adjust the class fields of the tasks and Storage class.

Adjusting the composition of the tasks requires additional
refactoring for less code duplication and adjustments to the
Storage class for storing/loading contents into the save file in
the hard drive of the user.

marclzh committed:
Merge pull request #3 from marclzh/C-Priority

Add Priority Feature

marclzh committed:
Change Elp's and user's profile picture, made text box cleaner

marclzh committed:
Create user guide and bug fixes

No user guide currently exists for the Elp chatbot.

It may be difficult to understand how to properly utilise the
chatbot without knowing the comments and syntax required.

Let's add a user guide for the project as a website to allow for
users to learn how to use the chatbot.

A github website is a clean and easy way to implement this, and can
be found at the URL: https://marclzh.github.io/ip/.

marclzh committed:
Adjust javadocs to be inline with guidelines

marclzh committed:
Fix user guide format

marclzh committed:
User guide fix

marclzh committed:
Fix YAML frontmatter bug

marclzh committed:
User guide header fix

marclzh committed:
Fix gui for linux
